### PR TITLE
Dashboard user statistics

### PR DESCRIPTION
**Pull request by abror9109**

### CODE REVIEW

File: app/Models/User.php [hunk 1]
```diff
@@ -51,4 +51,26 @@ public function skills(): BelongsToMany
...
+     * Scope a query to only include
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder $query
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeIdeaHolder($query)
+    {
+        return $query->where('user_type', '=', 'idea_holder');
+    }
 }
```
Comment (akbarali1) [inline app/Models/User.php:80]: Scopelar uchun modelda commentlarni qoldirish kerak

Comment (akbarali1) [reply app/Models/User.php:80]: Topilmagan funksiya bo`lmasligi uchun

**Review by akbarali1 — CHANGES_REQUESTED** [review]
user modelga developer va idealHolder degan scope uchun koemnt yozish kerak shunda u funksiyalarni ishlatganda undefined bermaydi

File: app/Models/User.php [hunk 2]
```diff
@@ -24,6 +25,8 @@
  * @property bool|null $is_admin
```
Comment (akbarali1) [inline app/Models/User.php:25]: bool|null menimcha xato 


?bool

bo`lishi kerak

File: app/Models/User.php [hunk 3]
```diff
@@ -24,6 +25,8 @@
...
  * @property Collection|Skill[] $skills
  * @property SmsVerification $sms
+ * @method static Builder developer()
```
Comment (akbarali1) [inline app/Models/User.php:29]: scope relation va property lar orasida joy tashlangan bo`lishi kerak

**Review by akbarali1 — CHANGES_REQUESTED** [review]
Komentlarni o`zgartirish kerak

**Review by akbarali1 — APPROVED** [review]

File: app/Models/User.php [hunk 4]
```diff
@@ -2,10 +2,11 @@
 
 namespace App\Models;
 
-// use Illuminate\Contracts\Auth\MustVerifyEmail;
```
Comment (Lanser0614) [inline app/Models/User.php:5]: cs fix i zapusk qilin

File: app/Models/User.php [hunk 5]
```diff
@@ -51,4 +55,26 @@ public function skills(): BelongsToMany
...
         return $this->belongsToMany(Skill::class, 'developer_skills_pivot');
     }
+
+    /**
+     * Scope a query to only include
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder $query
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeDeveloper($query): Builder
```
Comment (Lanser0614) [inline app/Models/User.php:65]: scopeWhereDeveloper

File: app/Models/User.php [hunk 6]
```diff
@@ -51,4 +55,26 @@ public function skills(): BelongsToMany
...
+        return $query->where('user_type', '=', UserTypeConstants::USER_TYPE_DEVELOPER);
+    }
+
+    /**
+     * Scope a query to only include
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder $query
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeIdeaHolder($query): Builder
```
Comment (Lanser0614) [inline app/Models/User.php:76]: scopeWhereIdeaHolder

**Review by Lanser0614 — COMMENTED** [review]
To'g'irlash kere

**Pull request merged**